webapp/src/pages/RegisterPage/RegisterPage.tsx

```diff
@@ -52,6 +52,8 @@ export default function RegisterPage() {
 
     if (formValues.username === '') {
       errMessage = 'Username can not be empty';
+    } else if (formValues.username === 'me') {
+      errMessage = 'Invalid username';
```

Comment:
suggestion: What do you think about changing the endpoints routes depending if we fetch info about the authenticated user or other users? (please regenerate the webapp generated code to test this)

Something like this:

<details>

```diff
diff --git a/transcendence-app/src/user/user.controller.ts b/transcendence-app/src/user/user.controller.ts
index aa7dfe0..81ee78e 100644
--- a/transcendence-app/src/user/user.controller.ts
+++ b/transcendence-app/src/user/user.controller.ts
@@ -68,14 +68,14 @@ export const AvatarFileInterceptor = LocalFileInterceptor({
   },
 });
 
-@Controller('users')
+@Controller()
 @UseGuards(AuthenticatedGuard)
 @ApiTags('users')
 @ApiForbiddenResponse({ description: 'Forbidden' })
 export class UserController {
   constructor(private userService: UserService) {}
 
-  @Patch()
+  @Patch('user')
   @ApiOkResponse({ description: 'Update the authenticated user', type: User })
   @ApiBadRequestResponse({ description: 'Bad Request' })
   @ApiUnprocessableEntityResponse({ description: 'Unprocessable Entity' })
@@ -93,13 +93,13 @@ export class UserController {
     return updatedUser;
   }
 
-  @Get('me')
+  @Get('user')
   @ApiOkResponse({ description: 'Get the authenticated user', type: User })
   getCurrentUser(@GetUser() user: User) {
     return user;
   }
 
-  @Get()
+  @Get('users')
   @ApiOkResponse({
     description: `Lists all users (max ${MAX_ENTRIES_PER_PAGE})`,
     type: [User],
@@ -116,7 +116,7 @@ export class UserController {
     return users;
   }
 
-  @Get(':userName')
+  @Get('users/:userName')
   @ApiOkResponse({ description: 'Get a user', type: UserResponseDto })
   @ApiNotFoundResponse({ description: 'Not Found' })
   @ApiBadRequestResponse({ description: 'Bad Request' })
@@ -134,7 +134,7 @@ export class UserController {
     return user;
   }
 
-  @Put('avatar')
+  @Put('user/avatar')
   @ApiConsumes('multipart/form-data')
   @ApiFile('file')
   @ApiProduces('image/jpeg')
@@ -168,7 +168,7 @@ export class UserController {
     return avatar;
   }
 
-  @Get('block/list')
+  @Get('user/block')
   @ApiOkResponse({
     description: 'List users blocked by the authenticated user',
     type: [User],
@@ -182,7 +182,7 @@ export class UserController {
     return blockedUsers;
   }
 
-  @Post('block/:userId')
+  @Post('user/block/:userId')
   @HttpCode(HttpStatus.NO_CONTENT)
   @ApiNoContentResponse({ description: 'Block a user' })
   @ApiBadRequestResponse({ description: 'Bad Request' })
@@ -194,7 +194,7 @@ export class UserController {
     return this.userService.addBlock(user.id, blockedUserId);
   }
 
-  @Delete('block/:userId')
+  @Delete('user/block/:userId')
   @HttpCode(HttpStatus.NO_CONTENT)
   @ApiNoContentResponse({
     description: 'Unblock a user',
diff --git a/webapp/src/pages/RegisterPage/RegisterPage.tsx b/webapp/src/pages/RegisterPage/RegisterPage.tsx
index 51a4f5f..d4c79cf 100644
--- a/webapp/src/pages/RegisterPage/RegisterPage.tsx
+++ b/webapp/src/pages/RegisterPage/RegisterPage.tsx
@@ -52,8 +52,6 @@ export default function RegisterPage() {
 
     if (formValues.username === '') {
       errMessage = 'Username can not be empty';
-    } else if (formValues.username === 'me') {
-      errMessage = 'Invalid username';
     } else if (formValues.email === '') {
       errMessage = 'Email can not be empty';
     } else if (formValues.fullName === '') {
diff --git a/webapp/src/shared/components/Button/Button.css b/webapp/src/shared/components/Button/Button.css
index 773855c..bf0da2a 100644
--- a/webapp/src/shared/components/Button/Button.css
+++ b/webapp/src/shared/components/Button/Button.css
@@ -6,6 +6,7 @@
   align-items: center;
   cursor: pointer;
   display: flex;
+  gap: 1rem;
   flex-direction: row;
   justify-content: center;
 }

```

</details>

Reply:
Done